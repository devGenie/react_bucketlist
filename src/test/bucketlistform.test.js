import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount } from 'enzyme';
import {stub, spy} from 'sinon';
import sinon from 'sinon';
import 'should';
import 'should-enzyme';
import toJson from 'enzyme-to-json';
import sinonStubPromise from 'sinon-stub-promise';

import BucketlistForm from '../components/bucketlist-form';
import localStorageMock from '../__mocks__/localStorage';

sinonStubPromise(sinon)
window.sessionStorage = localStorageMock;

describe('<BucketlistForm/>',()=>{
	let formMount;
	let fetchMock;
	let changeSpy;
	let submitSpy;

	beforeEach(()=>{
		fetchMock = stub(window,'fetch').returnsPromise().resolves({status:'success'});
		changeSpy = spy(BucketlistForm.prototype,'handleChange');
		submitSpy = spy(BucketlistForm.prototype,'handleSubmit');
		formMount = mount(<BucketlistForm />);
	});

	afterEach(()=>{
		fetchMock.restore();
		changeSpy.restore();
		submitSpy.restore()
	});

	it('can render correctly',()=>{
		expect(toJson(formMount)).toMatchSnapshot()
	});

	it('can submit data',()=>{
		formMount.find('form').simulate('submit',{
			preventDefault:()=>{}
		});
		expect(submitSpy.called).toBe(true)
	});

	it('can handle change event',()=>{
		formMount.find('form').find({name:'name'}).simulate('change',{
			target:{
				value:'changed'
			}
		});
		expect(changeSpy.called).toBe(true)
	})
})