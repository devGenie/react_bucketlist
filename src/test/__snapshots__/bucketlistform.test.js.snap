// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BucketlistForm/> can render correctly 1`] = `
<BucketlistForm>
  <div
    className="modal"
    id="activityModal"
  >
    <form
      onSubmit={[Function]}
    >
      <Loading
        status="hide"
      >
        <div
          className="progress hide"
        >
          <div
            className="indeterminate"
          />
        </div>
      </Loading>
      <div
        className="modal-content"
      >
        <h4
          className="header2"
        >
          New Bucketlist
        </h4>
        <FieldWrapper
          Label="Name"
        >
          <div
            className="input-field  "
          >
            <input
              className="validate "
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name of bucketlist"
              type="text"
              value=""
            />
            <label>
              Name
            </label>
          </div>
        </FieldWrapper>
        <FieldWrapper>
          <div
            className="input-field  "
          >
            <textarea
              className="validate "
              name="description"
              onChange={[Function]}
              placeholder="Description"
              value=""
            />
            <label />
          </div>
        </FieldWrapper>
        <FieldWrapper
          extraz="right"
        >
          <div
            className="input-field right"
          >
            <button
              className="btn "
              type="submit"
            >
              Create Bucketlist
            </button>
            <label />
          </div>
        </FieldWrapper>
      </div>
    </form>
  </div>
</BucketlistForm>
`;
